internal/grpc/services/publicstorageprovider/publicstorageprovider.go

```diff
@@ -334,3 +540,45 @@ func (s *service) pathFromToken(ctx context.Context, token string) (string, erro
 
 	return pathRes.Path, nil
 }
+
+// refFromToken returns the path for the publicly shared resource.
+func (s *service) refFromToken(ctx context.Context, token string) (*provider.Reference, error) {
+	driver, err := pool.GetPublicShareProviderClient(s.conf.DriverAddr)
```

Comment:
duplicated from pathFromToken except for the return statement that wraps it
remove and use pathFromToken instead